xilem_masonry/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+// These lints shouldn't apply to examples or tests.
+#![cfg_attr(not(test), warn(unused_crate_dependencies))]
+// These lints shouldn't apply to examples.
+#![warn(clippy::print_stdout, clippy::print_stderr)]
+// Targeting e.g. 32-bit means structs containing usize can give false positives for 64-bit.
+#![cfg_attr(target_pointer_width = "64", warn(clippy::trivially_copy_pass_by_ref))]
+// END LINEBENDER LINT SET
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+// TODO: Remove any items listed as "Deferred"
+#![cfg_attr(not(debug_assertions), allow(unused))]
+#![expect(
+    missing_debug_implementations,
+    reason = "Deferred: Noisy. Requires same lint to be addressed in Masonry"
+)]
+#![expect(clippy::missing_assert_message, reason = "Deferred: Noisy")]
+#![expect(elided_lifetimes_in_paths, reason = "Deferred: Noisy")]
+// https://github.com/rust-lang/rust/pull/130025
+#![expect(clippy::allow_attributes_without_reason, reason = "Deferred: Noisy")]
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use masonry::core::{
+    FromDynWidget, Properties, Widget, WidgetId, WidgetMut, WidgetOptions, WidgetPod,
+};
+
+use view::{Transformed, transformed};
+
+use crate::core::{
+    AsyncCtx, MessageResult, Mut, RawProxy, SuperElement, View, ViewElement, ViewId,
+    ViewPathTracker, ViewSequence,
+};
+
+pub use masonry::kurbo::{Affine, Vec2};
+pub use masonry::parley::Alignment as TextAlignment;
+pub use masonry::parley::style::FontWeight;
+pub use masonry::peniko::{Blob, Color};
+pub use masonry::widgets::{InsertNewline, LineBreaking};
+pub use masonry::{dpi, palette};
+pub use xilem_core as core;
+
+/// Tokio is the async runner used with Xilem.
+pub use tokio;
+
+mod any_view;
+mod one_of;
+mod property_tuple;
+
+pub mod style;
+pub mod view;
+pub use any_view::AnyWidgetView;
+pub use property_tuple::PropertyTuple;
+
+/// A container for a yet to be inserted [Masonry](masonry) widget
+/// to be used with Xilem.
+///
+/// This exists for two reasons:
+/// 1) The nearest equivalent type in Masonry, [`WidgetPod`], can't have
+///    [Xilem Core](xilem_core) traits implemented on it due to Rust's orphan rules.
+/// 2) `WidgetPod` is also used during a Widget's lifetime to contain its children,
+///    and so might not actually own the underlying widget value.
+///    When creating widgets in Xilem, layered views all want access to the - using
+///    `WidgetPod` for this purpose would require fallible unwrapping.
+#[expect(missing_docs, reason = "TODO - Document these items")]
+pub struct Pod<W: Widget + FromDynWidget + ?Sized> {
+    pub widget: Box<W>,
+    pub id: WidgetId,
+    /// The options the widget will be created with.
+    ///
+    /// If changing transforms of widgets, prefer to use [`transformed`]
+    /// (or [`WidgetView::transform`]).
+    /// This has a protocol to ensure that multiple views changing the
+    /// transform interoperate successfully.
+    pub options: WidgetOptions,
+    pub properties: Properties,
+}
```

Comment:
By the way, one of the things I'd like to do after this PR is merged is to split items like this into separate files. Right now Xilem basically implements half its features in its `lib.rs` file, which is terrible for readability.